File: 链表/160.相交链表.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        ListNode la = headA;
        ListNode lb = headB;
        while(la != lb){
            if(la==null){
                la=headB;
            }else{
                la=la.next;
            }
            if(lb==null){
                lb=headA;
            }else{
                lb=lb.next;
            }
        }
        return la;
    }
}

File: 链表/160.相交链表.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
package 链表

type ListNode struct {
	Val  int
	Next *ListNode
}

func getIntersectionNode(headA, headB *ListNode) *ListNode {
	cur1, cur2 := headA, headB

	for cur1 != cur2 {
		if cur1 != nil {
			cur1 = cur1.Next
		} else {
			cur1 = headB
		}
		if cur2 != nil {
			cur2 = cur2.Next
		} else {
			cur2 = headA
		}
	}
	return cur1

}
